### ISSUE TITLE
Fetch with as_dict=True raises a confusing error when the attribute list is specified.

### ISSUE THREAD

**Issue by ecobost** [datajoint/datajoint-python#391]
When using fetch(1), as_dict works only in certain cases. Some examples:

```
In [73]: reso.schema.jobs
Out[73]: 
*table_name    *key_hash      status       error_message  user           host           pid      connection_id  timestamp      key        error_stack   
+------------+ +------------+ +----------+ +------------+ +------------+ +------------+ +------+ +------------+ +------------+ +--------+ +------------+
__segmentation 30fff7922ea185 reserved                    ecobost@10.28. d6f4e9bc8ab8   2482     1221199        2017-11-13 21: <BLOB>     <BLOB>        
 (1 tuples)

In [74]: reso.schema.jobs.fetch1(as_dict=True)
Out[74]: 
OrderedDict([('table_name', '__segmentation'),
             ('key_hash', '30fff7922ea18543ea1f298418673640'),
             ('status', 'reserved'),
             ('key', rec.array([(1, 1, 1, 3, 3, 10577, 1)], 
                        dtype=[('scan_idx', 'O'), ('channel', 'O'), ('session', 'O'), ('segmentation_method', 'O'), ('field', 'O'), ('animal_id', 'O'), ('pipe_version', 'O')])),
             ('error_message', ''),
             ('error_stack', None),
             ('user', 'ecobost@10.28.2.115'),
             ('host', 'd6f4e9bc8ab8'),
             ('pid', 2482),
             ('connection_id', 1221199),
             ('timestamp', datetime.datetime(2017, 11, 13, 21, 51, 5))])

In [75]: reso.schema.jobs.fetch1('status', 'pid')
Out[75]: ('reserved', 2482)

In [76]: reso.schema.jobs.fetch1('status', 'pid', as_dict=True) # does not return a dict
Out[76]: ('reserved', 2482)

In [77]: type(_)
Out[77]: tuple
```

And with fetch, it just fails
```
In [86]: reso.schema.jobs.fetch('status', 'pid')
Out[86]: [array(['reserved'], dtype=object), array([2482])]

In [88]: reso.schema.jobs.fetch('status', 'pid', as_dict=True)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-79-e0ee9a14edc6> in <module>()
----> 1 reso.schema.jobs.fetch('status', 'pid', as_dict=True)

/usr/local/lib/python3.5/dist-packages/datajoint/fetch.py in __call__(self, *attrs, **kwargs)
    208                 list(to_dicts(result[self._relation.primary_key]))
    209                 if attribute is PRIMARY_KEY else result[attribute]
--> 210                 for attribute in attrs]
    211             ret = return_values[0] if len(attrs) == 1 else return_values
    212 

/usr/local/lib/python3.5/dist-packages/datajoint/fetch.py in <listcomp>(.0)
    208                 list(to_dicts(result[self._relation.primary_key]))
    209                 if attribute is PRIMARY_KEY else result[attribute]
--> 210                 for attribute in attrs]
    211             ret = return_values[0] if len(attrs) == 1 else return_values
    212 

TypeError: list indices must be integers or slices, not str
```

**Comment by dimitri-yatsenko** [comment]
Ok, the issue is that `as_dict` does not not work when you specify the attributes to return.  This is the correct behavior but the error message should be more clear.

Specifying the list of attributes distributes the result into variables:
```python
status, pid = reso.schema.jobs.fetch1('status', 'pid')
```
so that its output is always a list and `as_dict=True` cannot apply.  We should update the error message to clarify that.

If you want to get the result as a dict with specific fields, you need to project them first:
```python
reso.schema.jobs.proj('status', 'pid').fetch1() 
```

I am changing the issue title to reflect this problem more accurately.

**Comment by dimitri-yatsenko** [comment]
Furthermore, `fetch1` already returns a dict, so we will remove `as_dict` as a valid keyword argument from `fetch1`.

**Comment by dimitri-yatsenko** [comment]
Fixed in #377
